app/src/main/java/mozilla/lockbox/store/AccountStore.kt

```diff
@@ -106,66 +105,47 @@ open class AccountStore(
 
         securePreferences.putString(Constant.Key.firefoxAccount, Constant.App.testMarker)
 
-        profileSubject.onNext(FixedFxAProfile().asOptional())
+        profileSubject.onNext(testProfile.asOptional())
         syncCredentialSubject.onNext(FixedSyncCredentials(isNew).asOptional())
     }
 
     private fun populateAccountInformation(isNew: Boolean) {
         val profileSubject = profile as Subject
         val syncCredentialSubject = syncCredentials as Subject
+        val fxa = fxa ?: return
+        securePreferences.putString(Constant.Key.firefoxAccount, fxa.toJSONString())
 
-        fxa?.let { fxa ->
-            securePreferences.putString(Constant.Key.firefoxAccount, fxa.toJSONString())
+        fxa.getProfile()
+            .asSingle(Dispatchers.Default)
+            .delay(1L, TimeUnit.SECONDS)
+            .map { it.asOptional() }
+            .subscribe(profileSubject::onNext, this::pushError)
+            .addTo(compositeDisposable)
 
-            fxa.getProfile()
-                .asSingle(Dispatchers.Default)
-                .delay(1L, TimeUnit.SECONDS)
-                .subscribe({ profile ->
-                    profileSubject.onNext(profile.toFxAProfile().asOptional())
-                }, {
-                    dispatcher.dispatch(RouteAction.Dialog.NoNetworkDisclaimer)
-                })
-                .addTo(compositeDisposable)
-
-            fxa.getCachedOAuthToken(Constant.FxA.scopes)
-                .asMaybe(Dispatchers.Default)
-                .delay(1L, TimeUnit.SECONDS)
-                .subscribe({
-                    val syncCredentials = generateSyncCredentials(it, isNew)
-                    syncCredentialSubject.onNext(syncCredentials.asOptional())
-                }, {
-                    dispatcher.dispatch(RouteAction.Dialog.NoNetworkDisclaimer)
-                })
-                .addTo(compositeDisposable)
-        }
+        fxa.getAccessToken(Constant.FxA.oldSyncScope)
```

Comment:
Strictly speaking, obtaining a fresh access token is something that will happen with some regularity (whenever LoginsStorage "throws" a `SyncAuthInvalidException` error ... which should exist in `0.34` or higher).  Is it worth breaking this call out into its own method that can/will be wired up to some sort of "refresh token" action?

Reply:
I know I asked the question a leading way, but it is really a question.  To phrase it differently, should refreshing the token also refresh profile information?

I can see arguments for both sides.  The more I think about it, I slightly prefer refreshing both (if we're at this state, we've already pushed the radio to its most active state anyways, so might as well get as much data as we can).

Reply:
I'm curious what the intended behavior for this is on the client side, maybe @rfk or @vladikoff have thoughts?

Reply:
@linuxwolf's suggestion to refresh both sync token and profile info seems fine to me.  Longer-term we could try to get you a push notification whenever the profile info changes, if that seems worthwhile, since we're going to need to build that out for the reference-browser anyways.